cypress/integration/ResourcesForResearchers/index.js

```diff
@@ -0,0 +1,264 @@
+  cy.get('header > a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+And("search box is empty", () => {
+  cy.get('.searchbar__container input').should('contain.text', "");
+});
+
+And('results info text is {string}', (resultsInfo) => {
+  const regex = new RegExp(resultsInfo);
+  cy.get('[aria-label="Results count"] p').invoke('text').should('match', regex);
+});
+
+And('both pagers are displayed', () => {
+  cy.get('.r4r-results.r4r-DEFAULT>nav').should('have.length', 2);
+  cy.get('.r4r-results.r4r-DEFAULT>nav').each(($el) => {
+    expect('div r4r-pager__count  r4r-DEFAULT').to.exist;
+    expect('div r4r-pager__nav  r4r-DEFAULT').to.exist;
+  });
+
+});
+
+And('the results are displayed with each title containing a link', () => {
+  cy.get('[aria-label="search results"]').should('be.visible');
+  cy.get('[aria-label="search results"] > article>h2').first().should('be.visible').find('a').should('have.attr', 'href');
+});
+
+And('each result has a description text under the title', () => {
+  cy.get('[aria-label="search results"] > article>p').first().should('be.visible');
+});
+
+When('user clicks on {string}', (tool) => {
+  cy.get('.home-nav__main >div >div>a').contains(tool).click();
+});
+
+And('the filter displays the {string}', (filter) => {
+  cy.get('[aria-label="Selected Search Filters"] [tabindex] p').should('contain.text', filter);
+});
+
+And('the {string} box displays the filter {string}', (section, tool) => {
+  const regex = new RegExp(tool);
+  cy.get('.facet__box--tool-types').should('be.visible').find('h4').should('contain.text', section);
+  cy.get('.r4r-DEFAULT.facet__box.facet__box--tool-types > label[role="checkbox"] > .r4r-DEFAULT.filter__label').invoke('text').should('match', regex);
+
+});
+
+And('the {string} box is displayed for {string}', (section, eligibleToolType) => {
+  if (eligibleToolType == "Y") {
+    cy.get('.facet__box--tool-subtypes').should('be.visible');
+  }
+});
+
+And('the {string} box displays the correct {string} checked', (section, area) => {
+  const regex = new RegExp(area);
+  cy.get('.r4r-DEFAULT.facet__box> h4').eq[2].should('contain', section);;
+  cy.get('.r4r-DEFAULT.facet__box >label[role="checkbox"]>input[checked]').should('be.visible');
+
+});
+And('the {string} box is displayed', (section) => {
+  cy.get('.r4r-DEFAULT.results__facets >div').find('h4').should('contain.text', section);
+
+});
+
+When('user types {string} in the search bar', (searchText) => {
+  cy.get('.searchbar__container input').type(searchText);
+});
+
+When('user clicks on the magnifying glass icon', () => {
+  cy.get('.searchbar__container button').click();
+});
+
+And("search box displays the term {string}", (searchText) => {
+  cy.get('.searchbar__container input').should('have.attr', 'value', searchText);
+});
+
+When('user hits {string} key', (enterKey) => {
+  cy.get('.searchbar__container input').type('{enter}{enter}');
+  cy.wait(2000);
```

Comment:
better to remove hardcoded wait here and add a step in the feature file after this step
`And browser waits`

Reply:
done